File: python/test/t_InverseBoxCoxEvaluation_std.py
#! /usr/bin/env python

import openturns as ot
import openturns.testing as ott

ot.TESTPREAMBLE()

# Dimension of test
dimension = 3

# Lambda
lambdaVector = ot.Point(dimension)
lambdaVector[0] = 0.0
lambdaVector[1] = 0.5
lambdaVector[2] = 1

# Init of the function
myFunction = ot.InverseBoxCoxEvaluation(lambdaVector)
myFunction.setName("myInverseBoxCoxFunction")

# We check that for a Point with same values, the function is
# done by component
inPoint = ot.Point(dimension, 2.0)

# result point of the function
outPoint = myFunction(inPoint)
print("myFunction=", myFunction)
print(myFunction.getName(), "( ", inPoint, " ) = ", outPoint)

# Creation of a Sample
size = 10
inSample = ot.Sample(size, dimension)
for index in range(size):
    inSample[index] = ot.Point(dimension, (1.0 / (index + 1)))

# result of the function
outSample = myFunction(inSample)
print(myFunction.getName(), "( ", repr(inSample), " ) = ", repr(outSample))

# Function wrapper uses finite differences; check consistency
func = ot.Function(myFunction)
_ = func.gradient(inPoint)
_ = func.hessian(inPoint)
with ott.assert_raises(Exception):
    myFunction([1.0])
_ = myFunction.getParameter()
_ = myFunction.getParameterDescription()
myFunction.setParameter(myFunction.getParameter())
_ = myFunction.getMarginal(1)
_ = myFunction.getMarginal([0, 1])
study = ot.Study()
study.setStorageManager(ot.XMLStorageManager("invboxcox_eval.xml"))
study.add("f", func)
study.save()

File: lib/test/t_InverseBoxCoxEvaluation_std.cxx
//                                               -*- C++ -*-
/**
 * @brief The test file of class InverseBoxCoxEvaluation for standard methods
 *
 *  Copyright 2005-2023 Airbus-EDF-IMACS-ONERA-Phimeca
 *
 *  This library is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU Lesser General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This library is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU Lesser General Public License for more details.
 *
 *  You should have received a copy of the GNU Lesser General Public License
 *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
 *
 */
#include "openturns/OT.hxx"
#include "openturns/OTtestcode.hxx"

using namespace OT;
using namespace OT::Test;

int main(int, char *[])
{
  TESTPREAMBLE;
  OStream fullprint(std::cout);

  try
  {

    const UnsignedInteger dimension = 3;

    // Lambda
    Point lambda(dimension);
    lambda[0] = 0.;
    lambda[1] = 0.5;
    lambda[2] = 1;

    InverseBoxCoxEvaluation myFunction(lambda);
    myFunction.setName("inverseBoxCoxFunction");

    // In this test, we check that for a Point with same values, the function is done by component
    Point inPoint(dimension, 2.0);
    // result of the function
    Point outPoint = myFunction( inPoint );
    fullprint << "myFunction=" << myFunction << std::endl;
    fullprint << myFunction.getName() << "( " << inPoint << " ) = " << outPoint << std::endl;

    // Creation of a Sample
    const UnsignedInteger size = 10;
    Sample inSample(size, dimension);
    for (UnsignedInteger index = 0 ; index < size; ++index)
    {
      inSample[index] = Point(dimension, 1.0 / (index + 1));
    }

    // result of the function
    Sample outSample = myFunction( inSample );
    fullprint << myFunction.getName() << "( " << inSample << " ) = " << outSample << std::endl;

  }
  catch (TestFailed & ex)
  {
    std::cerr << ex << std::endl;
    return ExitCode::Error;
  }

  return ExitCode::Success;
}
